src/components/Savings/SavingsCard/SavingsCardTableMobile/index.tsx

```diff
@@ -9,95 +9,119 @@ import { ModalType } from 'components/Savings/SavingsModal'
 import { ModalButton } from 'components/Savings/SavingsModal/ModalButton'
 import { CellSC } from 'pages/gd/Portfolio/styled'
 import { LoadingPlaceHolder } from 'theme/components'
+import { NETWORK_LABEL } from 'constants/networks'
 import type { HeadingCopy } from '..'
+import { ChainId } from '@sushiswap/sdk'
+import { hasSavingsBalance } from 'functions'
 
 export const SavingsCardTableMobile = ({
     account,
     requiredChain,
-    hasBalance,
     headings,
     toggleModal,
 }: {
     account: string
-    requiredChain: number
-    hasBalance: boolean | undefined
+    requiredChain: ChainId
     headings: HeadingCopy
     toggleModal: (type?: ModalType) => void
 }): JSX.Element => {
     const { i18n } = useLingui()
     const { stats, error } = useStakerInfo(requiredChain, 10, account)
+    const [hasBalance, setHasBalance] = useState<boolean | undefined>(false)
```

Comment:
this could be a custom hook


```typescript
const useHasBalance = (account: string, requiredChain: ChainId) => {
  const [hasBalance, setHasBalance] = useState<boolean | undefined>(false)
  const { defaultEnv } = useGetEnvChainId(requiredChain)
  const provider = useReadOnlyProvider(requiredChain)

  useEffect(() => {
    if (account && provider) {
      hasSavingsBalance({ account, provider, defaultEnv }).then(setHasBalance)
    }
  }, [account, setHasBalance, provider, defaultEnv, requiredChain])

  return hasBalance
}
```